firstvoices/backend/views/data_views.py

```diff
@@ -44,25 +52,65 @@ class SitesDataViewSet(
     viewsets.GenericViewSet,
 ):
     http_method_names = ["get"]
-    serializer_class = SiteDataSerializer
-    pagination_class = None
+    serializer_class = DictionaryEntryDataSerializer
+    pagination_class = DictionaryEntryPaginator
+    renderer_classes = [SnakeCaseJSONRenderer, BrowsableAPIRenderer]
 
     def get_queryset(self):
-        sites = (
+        site = self.get_validated_site()
+        return DictionaryEntry.objects.filter(site__id__in=site).prefetch_related(
+            "part_of_speech",
+            "site",
+            Prefetch(
+                "related_audio",
+                queryset=Audio.objects.all()
+                .select_related(
+                    "original",
+                )
+                .prefetch_related(
+                    "speakers",
+                ),
+            ),
+            Prefetch(
+                "related_images",
+                queryset=Image.objects.all().select_related(
+                    "original",
+                ),
+            ),
+            Prefetch("site__alphabet_set", queryset=Alphabet.objects.all()),
+            "translation_set",
+            "categories",
+            "categories__parent",
+            "acknowledgement_set",
+            "note_set",
+        )
+
+    def list(self, request, *args, **kwargs):
+        queryset = utils.filter_by_viewable(request.user, self.get_queryset())
+        paginator = DictionaryEntryPaginator()
+        page = paginator.paginate_queryset(queryset, request)
+        serializer = self.get_serializer(page, many=True)
+
+        data = {}
+
+        site = (
             self.get_validated_site()
-            .select_related("language")
+            .select_related("language", "created_by", "last_modified_by")
             .prefetch_related(
-                Prefetch(
-                    "dictionaryentry_set",
-                    queryset=DictionaryEntry.objects.visible(self.request.user),
-                ),
-                "character_set",
-                "dictionaryentry_set__part_of_speech",
-                "alphabet_set",
-                "dictionaryentry_set__acknowledgement_set",
-                "dictionaryentry_set__note_set",
-                "dictionaryentry_set__categories",
-                "dictionaryentry_set__categories__parent",
+                "category_set__parent",
             )
         )
-        return sites
+
+        site_config_and_categories_json = SiteDataSerializer(
+            site[0], context={"request": request}
+        ).data
+
+        for key, value in site_config_and_categories_json.items():
+            data[key] = value
```

Comment:
This is totally fine, but just a heads up that this kind of thing is a good use for the python `...` magic, like `data = { ...other_dictionary }`